pkg/compression/compression.go

```diff
@@ -35,32 +36,85 @@ func XzDecompressor(r io.Reader) (io.ReadCloser, error) {
 	return ioutil.NopCloser(r), nil
 }
 
+// compressorFunc returns the compressed stream, given a compressed stream.
+// The caller must call Close() on the compressed stream (even if the compressed input stream does not need closing!).
+type compressorFunc func(io.Writer, int) (io.WriteCloser, error)
+
+// GzipCompressor is a CompressorFunc for the gzip compression algorithm.
+func GzipCompressor(r io.Writer, level int) (io.WriteCloser, error) {
```

Comment:
Given the new API compatibility constraints, I’d prefer making as few parts of the API public as possible. We can always make them public in the future if it turns out to be necessary, the reverse is now problematic.

AFAICS we don’t need the individual compression handlers to be public. (Nor, ultimately, `ZstdDecompressor`, although in that case arguably it’s cleaner to keep the shape of the API the same for all compression algorithms.)